kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/interceptor/SpecInterceptorPipeline.kt

```diff
@@ -48,6 +50,8 @@ internal class SpecInterceptorPipeline(
 
    private fun createPipeline(): List<SpecInterceptor> {
       return listOfNotNull(
+         CoroutineNameInterceptor,
+         SpecContextElementInteceptor,
```

Comment:
Could we combine these two into an interceptor that adds any type of `CoroutineContext` at spec-level and then use that interceptor for `CoroutineName` and `SpecContextElement`?